### PR TITLE
workaround for num_cpus returning 0

### PR DESCRIPTION
**Pull request by AbhilashBalaji**
#1023 , This is my first pr with rust , do let  me know if  there are any issues :)

### CODE REVIEW

**Comment by HeavySnowJakarta** [comment]
> #1023 , This is my first pr with rust , do let  me know if  there are any issues :)

Want issues?Come here and have a look(huaji,a kind of China meme): https://github.com/ogham/exa/issues/1023 

A developer has told me there is something wrong with num_cpus for returning 0 on my emulator environment,and I guess you may be interested in having a look.

**Comment by HeavySnowJakarta** [comment]
> rust

First I need to appreciate it for your attention to this problem.What's more,I want to know more details about what the change of this fork is and whether&when I may try this fork.

File: src/output/details.rs [hunk 1]
```diff
@@ -147,7 +147,12 @@ impl<'a> AsRef<File<'a>> for Egg<'a> {
 
 impl<'a> Render<'a> {
     pub fn render<W: Write>(mut self, w: &mut W) -> io::Result<()> {
-        let mut pool = Pool::new(num_cpus::get() as u32);
+        let mut n_cpus = num_cpus::get() as u32;
+        n_cpus = match n_cpus {
+            0 => 1,
+            _ => n_cpus,
+        };
```
Comment (ariasuni) [inline src/output/details.rs:154]: You can slightly improve the code by doing that:

```suggestion
        let n_cpus = match num_cpus::get() as u32 {
            0 => 1,
            _ => n_cpus,
        };
```

**Review by ariasuni — CHANGES_REQUESTED** [review]

**Comment by ariasuni** [comment]
> First I need to appreciate it for your attention to this problem.What's more,I want to know more details about what the change of this fork is and whether&when I may try this fork.

In some cases, exa crashes because we’re unable to determine the number of CPU for parallelization. Instructions on how to build exa are on the README, and you can find information on how to get locally the branch of a GitHub pull request with Git on the web.

File: src/output/details.rs (hunk 1 above)
Comment (AbhilashBalaji) [reply src/output/details.rs:154]: This is what I initially tried , but i get   this error , `n_cpus` is not defined yet. any other suggestions ?
```
error[E0425]: cannot find value `n_cpus` in this scope
   --> src/output/details.rs:152:18
    |
152 |             _ => n_cpus,
    |                  ^^^^^^ not found in this scope

```

Comment (ariasuni) [reply src/output/details.rs:154]: Ah yeah sorry my code is incorrect, sorry. This should work:

```suggestion
        let n_cpus = match num_cpus::get() as u32 {
            0 => 1,
            n => n,
        };
```

**Pull request merged**